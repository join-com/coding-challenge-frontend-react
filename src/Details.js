import React from "react";
import Fetcher from "./components/Fetcher";
import Loader from "./components/Loader";
import Describe from "./components/Describe";

import Bikevexapi from "./components/Bikevexapi";
import ErrorBoundary from "./components/ErrorBoundary";


class Details extends React.Component {

	render() {

		let req = Bikevexapi.incident;
		let IncidentDetails = Loader(Describe);
		let uri = `${req.path}/${this.props.id}`;
		return (
			<ErrorBoundary>
				<Fetcher path={uri} name={req.name}>
					{(props) => {
						// const { data, code } = props;
						// if (data) {
						// const { id, title, description, address, occured_at, media, type, source } = data[0];
						return (<IncidentDetails {...props} />);
						// } else {
						// 	return <Describe  {...props}></Describe>
						// }

					}}
				</Fetcher>
			</ErrorBoundary >
		);
	}
}

export default Details;