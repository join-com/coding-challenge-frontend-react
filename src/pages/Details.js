import React from 'react'

export default function Details({ match }) {
  return <h1>Id {match.params.id}!</h1>
}
