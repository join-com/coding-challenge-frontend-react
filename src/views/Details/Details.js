import React, { Component } from 'react'

class Details extends Component {
  render () {
    return (
      <div>
        Details view
      </div>
    )
  }
}

export default Details
