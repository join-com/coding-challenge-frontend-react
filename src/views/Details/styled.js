import styled from 'styled-components'

const Container = styled.div`
  max-width: 1140px;
  margin: auto;
  padding: 10px;
`

const HeaderContainer = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 20px 0;
`

export { Container, HeaderContainer }
