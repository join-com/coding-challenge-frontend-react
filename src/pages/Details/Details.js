import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import { withRouter } from 'react-router-dom'

import Case from './Case/Case'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  // width: 90%;
  // margin: 50px;
`

const Back = styled.a`
  color: blue;
`

function Details({ history }) {
  const [data, setData] = useState({ incidents: [] })
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    async function fetchData() {
      const pathnameSplitted = window.location.pathname.split('incident/')
      if (pathnameSplitted.length < 2) {
        // FIXME redirect to 404
        throw new Error('Redirect to 404')
      }

      const id = pathnameSplitted[1]

      if (loading) {
        const url = `https://bikewise.org/api/v2/incidents/${id}`
        const response = await window.fetch(url)
        const results = await response.json()
        setData(results)
        setLoading(true)
      }
    }
    fetchData()
  }, [])
  const item = data.incident
  return (
    <Wrapper>
      <Back onClick={history.goBack}>Back</Back>
      {item && <Case key={item.id} item={item} />}
    </Wrapper>
  )
}

export default withRouter(Details)
