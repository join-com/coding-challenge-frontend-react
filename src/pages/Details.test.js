import React from 'react'
import { shallow } from 'enzyme'
import Details from './Details'

it('renders the Details', () => {
  const wrapper = shallow(<Details match={{ params: { id: 1 } }} />)
  expect(wrapper).toMatchSnapshot()
})
