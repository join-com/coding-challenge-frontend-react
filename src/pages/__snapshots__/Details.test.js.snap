// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Details 1`] = `
<h1>
  Id 
  1
  !
</h1>
`;
